kratos/geometries/tetrahedra_3d_4.h

```diff
@@ -1070,6 +1070,45 @@ template<class TPointType> class Tetrahedra3D4 : public Geometry<TPointType>
         return rResult;
     }
 
+    /**
+    * @brief Checks if given point in local space coordinates of this geometry
+    *        is inside the geometry boundaries.
+    * @param rPointLocalCoordinates the point on the geometry,
+    *        which shall be checked if it lays within
+    *        the boundaries.
+    * @param Tolerance the tolerance to the boundary.
+    * @return -1 -> failed
+    *          0 -> outside
+    *          1 -> inside
+    *          2 -> on the boundary
+    */
+    virtual int IsInsideLocalSpace(
+        const CoordinatesArrayType& rPointLocalCoordinates,
+        const double Tolerance = std::numeric_limits<double>::epsilon()
+    ) const override
+    {
+        int is_inside = -1;
+        if (rPointLocalCoordinates[0] < 0.0-Tolerance 
+            || rPointLocalCoordinates[1] < 0.0-Tolerance
+            || rPointLocalCoordinates[2] < 0.0-Tolerance
+            || rPointLocalCoordinates[0] > 1.0+Tolerance 
+            || rPointLocalCoordinates[1] > 1.0+Tolerance
+            || rPointLocalCoordinates[2] > 1.0+Tolerance)
+        {
+            is_inside = 0; // is outside
+        } else if (rPointLocalCoordinates[0] > 0.0
+            && rPointLocalCoordinates[0] < 1.0
+            && rPointLocalCoordinates[1] > 0.0 
+            && rPointLocalCoordinates[1] > 0.0
+            && rPointLocalCoordinates[2] > 0.0
+            && rPointLocalCoordinates[2] < 1.0)
+            is_inside = 1; // is inside
+        else {
+            is_inside = 2; // is on boundary
+        }
+        return is_inside;
```

Comment:
Are you sure that the local coordinate [0.51,0.51,0.51] is not outside? Your implementation says it is inside.